content/spire/try/oidc_federation_aws.md

```diff
@@ -0,0 +1,399 @@
+23456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TO
+KEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789J
+WT-TOKEN0123456789JWT-TOKEN012
+```
+
+This environmental variable will be used by the AWS `aws sts assume-role-with-web-identity` command later.
+
+3. Save the JWT to a file for use by the AWS `aws s3 cp` command later.
+
+```console
+echo JWT=0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN01234
+56789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN
+0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-
+TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN012345678
+9JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123
+456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKE
+N0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT
+-TOKEN0123456789JWT-TOKEN012 > token
+```
+
+4. Optionally, test that you can retrieve an STS token by running the following command:
+
+```console
+aws sts assume-role-with-web-identity --role-arn arn:aws:iam::012345678901:role/oidc-role \
+--role-session-name mys3 --web-identity-token $JWT
+```
+
+If successful, the command will return JSON output similar to the following.
+
+```json
+{
+    "Credentials": {
+        "AccessKeyId": "0123456789ABCDEFGHIJ",
+        "SecretAccessKey": "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123",
+        "SessionToken": "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRS
+TUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789A
+BCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRS
+TUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789A
+BCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRS
+TUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789A
+BCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXY=",
+        "Expiration": "2019-10-11T21:59:06Z"
+    },
+    "SubjectFromWebIdentityToken": "spiffe://spire-oidc.example.org/oidc-test-workload",
+    "AssumedRoleUser": {
+        "AssumedRoleId": "0123456789ABCDEFGHIJK:mys3",
+        "Arn": "arn:aws:sts::012345678901:assumed-role/oidc-role/mys3"
+    },
+    "Provider": "arn:aws:iam::012345678901:oidc-provider/spire-oidc.example.org",
+    "Audience": "mys3"
+}
+```
+
+5. Access S3 using the JWT that you retrieved earlier. In this example, we're copying the file `s3://oidc-federation-test/test.txt` to our local environment. You would run the following command on the instance where you've installed the SPIRE Agent, specifying the ARN of the IAM role that you created earlier:
```

Comment:
How do I get the ARN of the IAM role? (remember the reader may quite likely not be that familiar with AWS)